libminifi/src/processors/TailFile.cpp

```diff
@@ -287,14 +288,36 @@ void TailFile::onTrigger(core::ProcessContext *context, core::ProcessSession *se
     std::string baseName = _currentTailFileName.substr(0, found);
     std::string extension = _currentTailFileName.substr(found + 1);
 
-    if (!this->_delimiter.empty()) {
-      char delim = this->_delimiter.c_str()[0];
+    if (!delimiter_.empty()) {
+      char delim = delimiter_.c_str()[0];
+      if (delim == '\\') {
+        if (delimiter_.size() > 1) {
+          switch (delimiter_.c_str()[1]) {
+            case 'r':
+              delim = '\r';
+              break;
+            case 't':
+              delim = '\t';
+              break;
+            case 'n':
+              delim = '\n';
+              break;
+            case '\\':
+              delim = '\\';
+              break;
+            default:
+              // previous behavior
```

Comment:
Maybe a log statement here?
This seems to be a configuration error as the delimiter string is longer than 1, but it doesn't contain any escaped char we support.

Reply:
That is the purpose of the log that was added below this.